pkg/proxy/server.go

```diff
@@ -703,10 +713,15 @@ func (s *Server) getTelnetConn() (srvConn *srvconn.TelnetConnection, err error)
 	asset := s.connOpts.authInfo.Asset
 	platform := s.connOpts.authInfo.Platform
 
-	protocolSetting := platform.GetProtocol(protocol)
-	usernamePrompt := strings.TrimSpace(protocolSetting.Setting.TelnetUsernamePrompt)
-	passwordPrompt := strings.TrimSpace(protocolSetting.Setting.TelnetPasswordPrompt)
-	successPrompt := strings.TrimSpace(protocolSetting.Setting.TelnetSuccessPrompt)
+	usernamePrompt := ""
+	passwordPrompt := ""
+	successPrompt := ""
+	if platfromProtocol, ok := platform.GetProtocolSetting(protocol); ok {
+		protocolSetting := platfromProtocol.GetSetting()
+		usernamePrompt = strings.TrimSpace(protocolSetting.TelnetUsernamePrompt)
+		passwordPrompt = strings.TrimSpace(protocolSetting.TelnetPasswordPrompt)
+		successPrompt = strings.TrimSpace(protocolSetting.TelnetSuccessPrompt)
+	}
 
 	if usernamePrompt != "" {
 		usernamePattern, err1 := regexp.Compile(usernamePrompt)
```

Comment:
The given code appears to be incomplete, missing context, and lacks details about its usage. Please provide more information so that I can assist you with checking the code and highlighting the necessary changes or recommendations for improvement.